moped-editor/src/views/projects/projectView/ProjectSummary/ProjectSummaryKnackDataTrackerSync.js

```diff
@@ -65,33 +82,62 @@ const ProjectSummaryKnackDataTrackerSync = ({
    * because if you update the project number field, even with the same, extant number, Knack returns an error.
    * @returns string
    */
-  const buildBody = () => {
+  const buildBody = signalIds => {
     let body = {};
 
-    const field_map = {
-      field_3998: "project_id",
-      field_3999: "project_name",
-      field_4000: "current_status",
-    };
+    const knackFieldsRegEx = /REACT_APP_KNACK_DATA_TRACKER_(\w+)_FIELD/;
+    const fieldMap = {};
+    Object.keys(process.env)
+      .filter(envVariable => {
+        return envVariable.match(knackFieldsRegEx);
+      })
+      .map(envVariable => {
+        return envVariable.match(knackFieldsRegEx);
+      })
+      .forEach(regExResult => {
+        fieldMap[process.env[regExResult[0]]] = regExResult[1].toLowerCase();
+      });
 
-    Object.keys(field_map).forEach(element => {
-      if (project.currentKnackState[element] !== project[field_map[element]]) {
-        body[element] = project[field_map[element]];
-      }
+    Object.keys(fieldMap).forEach(element => {
+      body[element] = project[fieldMap[element]];
```

Comment:
This is the consumption of the ENV variables mentioned above with the suspect naming scheme.

Thanks to @chiaberry for her advice on how to clean up the filter().map().forEach() chaining.